internal/services/instances.go

```diff
@@ -616,3 +606,48 @@ func getEpinioValues(serviceName, catalogServiceName string) (string, error) {
 
 	return "\n" + string(yamlData), nil
 }
+
+func setServiceStatusAndCustomValues(service *models.Service,
```

Comment:
I'm not completelyh sure about this func. I would prefer to keep the two updates separate, also because the second is not returning any error, and it seems that the update of the custom values is done only once (the func is called with the last argument always as `nil` except in the first case.

i.e. the two will be:

```go
func setServiceStatus(
	ctx context.Context,
	logger logr.Logger,
	cluster *kubernetes.Cluster,
	service *models.Service,
	serviceRelease *helmrelease.Release,
) error {
	service.Status = models.ServiceStatusUnknown

	serviceStatus, err := helm.Status(ctx, logger, cluster, serviceRelease)
	if err != nil {
		return errors.Wrap(err, "calculating helm release status")
	}
	service.Status = NewServiceStatusFromHelmRelease(serviceStatus)
	return nil
}
```

```go
func setServiceCustomValues(
	service *models.Service,
	serviceRelease *helmrelease.Release,
	settings map[string]models.ChartSetting,
) {
	if len(settings) > 0 {
		customized := models.ChartValueSettings{}
		configValues := chartutil.Values(serviceRelease.Config)

		for key := range settings {
			customValue, err := configValues.PathValue(key)
			if err != nil {
				// Not found - This custom value was not customized by the user.
				// That is ok. Nothing to report.
				continue
			}
			customValueAsString := fmt.Sprintf("%v", customValue)
			customized[key] = customValueAsString
		}

		if len(customized) > 0 {
			service.Settings = customized
		}
	}
}
```

Reply:
Of the four places calling the function ...
2 places are for helm controller backward compat, to be ripped out soon.
1 place is for service listing, where I decided the CVs would not be needed.

Reply:
Sure, can split the code.

Reply:
Your new `setServiceStatus` does not retrieve the release. That is the nice thing about the original. It pulls the release, then does status and CVs. And if there is no release it does nothing.

In your split the release retrieval either goes into the caller (i.e. replicated), or into the service status helper.

Reply:
Ah, I see! Good to know that this is going to be simplified a bit. Thanks!

Reply:
Ok, resolved to keep it as is now, in huddle.